src/main/java/org/jenkinsci/plugins/workflow/support/concurrent/DirectExecutor.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2007 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.jenkinsci.plugins.workflow.support.concurrent;
+
+import com.google.common.annotations.GwtCompatible;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import java.util.concurrent.Executor;
+
+/**
+ * An {@link Executor} that runs each task in the thread that invokes {@link Executor#execute
+ * execute}.
+ */
+@GwtCompatible
+@Restricted(NoExternalUse.class)
```

Comment:
Taken without change from Guava 30.1. Used only by the vendored code. Marked as restricted to prevent any further usages.

Reply:
should this be added as a comment in the source?

Reply:
I think the commit message should cover the provenance well enough:

```
commit ec98115 Add MoreExecutors#directExecutor() implementation from Guava 30.1
```

Reply:
Though `git blame` will not show that if this PR is squash-merged, as @car-roll normally seems to do.

Reply:
yeah, i noticed that line too. I was going to make a special exception :)